templates/terraform/examples/base_configs/iam_test_file.go.erb

```diff
@@ -0,0 +1,183 @@
+<%= lines(autogen_notice :go) -%>
+
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+<% example = object.examples.reject(&:skip_test)
+         .reject { |e| @api.version_obj_or_closest(version) < @api.version_obj_or_closest(e.min_version) }
+         .first -%>
+<% resource_name = product_ns + object.name -%>
+<% 
+individual_url = object.self_link_url
+params = individual_url.scan(/({{)(\w+)(}})/).map { |arr| arr[1] }
+-%>
+<% 
+if @config.legacy_name.nil?
+	terraform_name = "google_" + (product_ns + object.name).underscore
+else
+	terraform_name = "google_" + @config.legacy_name + '_' + object.name.underscore
+end 
+-%>
+func TestAcc<%= resource_name -%>IamBindingGenerated(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(10),
+		"role":          "roles/editor",
+	}
+
+<% import_url = individual_url.gsub(/({{)(\w+)(}})/, '%s').gsub(object.__product.base_url, '') -%>
+<% import_str = Array.new(params.length, '%s').join('/') -%>
+<% import_qualifiers = [] -%>
+<% params.each do |param| -%>
+	<% if param == 'project' -%>
+		<% import_qualifiers.push('getTestProjectFromEnv()') -%>
+	<% elsif param == 'zone' -%>
+		<% import_qualifiers.push('getTestZoneFromEnv(t)') -%>
+	<% elsif param == 'region' || param == 'location' -%>
+		<% import_qualifiers.push('getTestRegionFromEnv(t)') -%>
+	<% end -%>
+<% end -%>
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheck<%= resource_name -%>Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAcc<%= resource_name -%>IamBinding_basicGenerated(context),
+			},
+			{
+				ResourceName:      "<%= terraform_name -%>_iam_binding.foo",
+				ImportStateId:     fmt.Sprintf("<%= import_url -%> roles/editor", <%= import_qualifiers.join(', ') -%>, <%= object.iam_policy.primary_resource_name -%>),
```

Comment:
Ah, found out why we need this. The resource id and the import id are slightly different:

Resource ID:

```
projects/ci-test-project-188019/topics/test-topic-iam-bcrwgggdu3/roles/pubsub.publisher/serviceaccount:test-topic-iam-ruco3r060w@ci-test-project-188019.iam.gserviceaccount.com
```

Import ID:

```
projects/ci-test-project-188019/topics/test-topic-iam-bcrwgggdu3 roles/pubsub.publisher serviceAccount:test-topic-iam-ruco3r060w@ci-test-project-188019.iam.gserviceaccount.com\
```

Ideally, we'd store the `id` as a valid import ID, but that probably shouldn't be done in this PR.

Reply:
Instead of modifying `api.yaml`, to get the example's GCP-side resource name, we could add an optional field to the example object, or look for a value in `vars` like `primary_resource_name` (replacing values like https://github.com/GoogleCloudPlatform/magic-modules/blob/master/products/pubsub/terraform.yaml#L23)

Reply:
Moving to the terraform example object